protoc-gen-hack/namespace.go

```diff
@@ -43,39 +45,59 @@ type Namespace struct {
 	Fqn      string
 	Names    *Names
 	Children map[string]*Namespace
+	// fully qualified name prefixed with custom namespace
+	cFqn string
 }
 
 func NewEmptyNamespace() *Namespace {
-	return newNamespace(nil, "")
+	customNs := ""
+	return newNamespace(nil, "", &customNs)
 }
 
-func newNamespace(parent *Namespace, myName string) *Namespace {
+// Builds a namespace tree with the given fully qualified name and custom fully qualified name.
+func newNamespace(parent *Namespace, myName string, customName *string) *Namespace {
```

Comment:
Go doesn't have optional parameters so had to use a variadic(reference type) parameter `customName`.